src/hooks/transactions/usePayload.ts

```diff
@@ -32,30 +32,33 @@ interface Input {
 export const usePayload = ({ call, weight }: Input) => {
   const { createType } = useApiCallsContext();
   const {
-    sourceChainDetails: { chain: sourceChain }
+    sourceChainDetails: { chain: sourceChain },
+    targetChainDetails: {
+      apiConnection: { api: targetApi, isApiReady: isTargetApiReady }
+    }
   } = useSourceTarget();
   const { dispatchTransaction } = useUpdateTransactionContext();
   const { account } = useAccountContext();
 
   const payloadCallback = useCallback(() => {
-    if (!account || !call || !weight) {
+    if (!account || !call || !weight || !isTargetApiReady) {
```

Comment:
Do we really need the `isTargetApiReady` check? I mean - what is `targetApi.consts.system.version.specVersion` supposed to contain if the API is not ready?

Reply:
my understanding is that all child objects for the api are undefined until the api is ready.
@wirednkod  what do you think?

Reply:
If api is not ready you ll probably get an error saying `cannot find const of undefined`

Reply:
cool .. then is needed. Thanks!

Reply:
No, not really. It will error out when trying to set up the hook, because you pass `specVersion` in the dependency list.